app/main.py

```diff
@@ -1,34 +1,123 @@
+                decks.append(Deck(start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                decks.append(Deck(row, start_col))
+        return decks
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck or None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
+
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
+            if all(not d.is_alive for d in self.decks):
+                self.is_drowned = True
+            return True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self,
+                 ships: list[tuple[tuple[int, int], tuple[int, int]]]
+                 ) -> None:
+        self.size = 10
+        self.field = [["~" for _ in range(self.size)]
+                      for _ in range(self.size)]
+        self.ships = [Ship(*ship) for ship in ships]
+        self._place_ships()
+
+    @staticmethod
+    def _validate_field(ships: list) -> None:
+        if len(ships) != 10:
+            raise ValueError("There must be 10 ships.")
+        for ship in ships:
+            other_ships = [
+                end for temp in ships if temp != ship for end in temp
+            ]
+            for end in ship:
+                neighbours = [
+                    (end[0] - 1, end[1] - 1),
+                    (end[0] - 1, end[1]),
+                    (end[0] - 1, end[1] + 1),
+                    (end[0], end[1] - 1),
+                    (end[0], end[1] + 1),
+                    (end[0] + 1, end[1] - 1),
+                    (end[0] + 1, end[1]),
+                    (end[0] + 1, end[1] + 1),
+                ]
+
+                for neighbour in neighbours:
+                    if neighbour in other_ships:
+                        raise ValueError("The ships must not be adjacent.")
+
+        deck_size = []
+        for ship in ships:
+            deck_size.append(
+                (ship[1][0] - ship[0][0]) + (ship[1][1] - ship[0][1]) + 1
+            )
+        if sorted(deck_size) != [1, 1, 1, 1, 2, 2, 2, 3, 3, 4]:
+            raise ValueError("Wrong ship size configuration.")
```

Comment:
The `_validate_field` method is responsible for ensuring the correct number and type of ships, and that ships are not adjacent. Double-check the logic for checking ship adjacency and size configuration to ensure it works as expected. This is crucial for maintaining the integrity of the game setup.